keps/sig-multicluster/4322-cluster-inventory/README.md

```diff
@@ -524,12 +524,193 @@ minimum kubelet version, maximum kubelet version, and enabled featureset version
+* The list is dynamic but does not change much unless there is a node auto provisioner.
+
+Here is an example of a type.node.k8s.io ClusterProperty
+```yaml
+apiVersion: about.k8s.io/v1
+kind: ClusterProperty
+metadata:
+  name: type.node.k8s.io
+  labels:
+    multicluster.x-k8s.io/clusterProfile: true
+Spec:
+  value: ["g6.xlarge","Standard_NC48ads_H100","m3-ultramem-32","largeCPU","smallMem"]
+```
+Here is the corresponding ClusterProfile API
+```yaml
+properties:
+   - name: type.node.k8s.io
+     value: ["g6.xlarge","Standard_NC48ads_H100","m3-ultramem-32","largeCPU","smallMem"]
+```
+
+##### distribution.node.k8s.io
+It contains the node count for each node type in the k8s clusters
+* Each element in the list is a key-value pair representing the type of the node and its count.
+* The value is dynamic but should not change fast.
+
+Here is an example of a distribution.node.k8s.io ClusterProperty
+```yaml
+apiVersion: about.k8s.io/v1
+kind: ClusterProperty
+metadata:
+  name: distribution.node.k8s.io
+  labels:
+    multicluster.x-k8s.io/clusterProfile: true
+Spec:
+  value: [{"largeCPU":5}, {"smallMem":5}]
+```
+Here is the corresponding ClusterProfile API
+```yaml
+properties:
+   - name: distribution.node.k8s.io
+     value: [{"largeCPU":5}, {"smallMem":5}]
+```
+
+##### metrics-endpoints.k8s.io
+It contains an array of the type, url pair that one can query the metrics (for example, a Prometheus or Grafana endpoint for PromQL) of the cluster
+* Each element in the listing is a key-value pair representing the type of the metrics endpoint and its url.
+    * It could contain IP addresses too
+* The list is dynamic but should not change much.
+
+Here is an example of a metrics-endpoints.k8s.io ClusterProperty
+```yaml
+apiVersion: about.k8s.io/v1
+kind: ClusterProperty
+metadata:
+  name: metrics-endpoints.k8s.io
+  labels:
+    multicluster.x-k8s.io/clusterProfile: true
+Spec:
+  value: [{"Prometheus":"100.3.3.4:9990"}, {"Grafana":"example.grafana.io:9990"}]
+```
+Here is the corresponding ClusterProfile API
+```yaml
+properties:
+   - name: metrics-endpoints.k8s.io
+     value: [{"Prometheus":"100.3.3.4:9990"}, {"Grafana":"example.grafana.io:9990"}]
+```
+
+##### group.customResource.k8s.io
+It contains an array of custom resources definitions (CRDs) group that this cluster supports.
+* Each element in the list is a string representing the group name of the CRD.
+* The list is dynamic but should not change much.
+
+Here is an example of a group.customResource.k8s.io ClusterProperty
+```yaml
+apiVersion: about.k8s.io/v1
+kind: ClusterProperty
+metadata:
+  name: group.customResource.k8s.io
+  labels:
+    multicluster.x-k8s.io/clusterProfile: true
+Spec:
+  value: ["argoproj.io","kubeflow.org","istio.io","volcano.sh"]
+```
+Here is the corresponding ClusterProfile API
+```yaml
+properties:
+   - name: group.customResource.k8s.io
```

Comment:
should we include the version? consumer may need to know what's the latest version installed.

Reply:
sure, but CRD version might not show the real version. i.e argo CD version has been stuck in v1alpha1 forever.